queue-dir/queue.py

```
@@ -0,0 +1,68 @@
+from node import Node
+
+class Queue:
+    """
+    This class makes a queue data structure
+    """
+    def __init__(self, iter = []):
+        """
+        method which defines the newly created stack
+        """
+        self._back = None
+        self._front = None
+        self._size = 0
+
+        for i in reversed(iter):
+            self.enqueue(i)
+
+    def __len__(self):
+        """
+        returns current number of nodes in queue
+        """
+        return self._size        
+
+    def enqueue(self, val):
+        """
+        method which adds a value to the back of the queue
+        """
+        self._back = Node(val, self._back)
+        self._size += 1
+        return self._back
+
+    def dequeue(self):
+        """
+        iterates through the queue to find front node, then removes/returns it
+        """
+        ## need to test for 0 and 1 node
+        if self._size == 0:
+            print('Queue is empty, nothing to dequeue')
+            return False
```

Instead of returning `False` here you might want to raise an exception instead. What if the user is storing booleans in the queue?